src/3d/materials/qgsabstractmaterialsettings.h

```diff
@@ -170,8 +178,60 @@ class _3D_EXPORT QgsAbstractMaterialSettings SIP_ABSTRACT
      * Adds parameters from the material to a destination \a effect.
      */
     virtual void addParametersToEffect( Qt3DRender::QEffect *effect ) const = 0;
+
+    //! Data definable properties.
+    enum Property
+    {
+      Diffuse, //!< Diffuse color
+      Ambient, //!< Ambient color (phong material)
+      Warm, //!< Warm color (gooch material)
+      Cool,//!< Cool color (gooch material)
+      Specular //!< Specular color
+    };
+
+    /**
+     * Sets the material property collection, used for data defined overrides.
+     * \since QGIS 3.18
+     */
+    void setDataDefinedProperties( const QgsPropertyCollection &collection );
+
+    /**
+    * Returns the symbol material property collection, used for data defined overrides.
+    * \since QGIS 3.18
+    */
+    QgsPropertyCollection dataDefinedProperties() const;
+
+    /**
+    * Returns a reference to the material properties definition, used for data defined overrides.
+    * \since QGIS 3.18
+    */
+    const QgsPropertiesDefinition  &propertyDefinition() const;
```

Comment:
```suggestion
    const QgsPropertiesDefinition  &propertyDefinitions() const;
```

:wink:

needs to match all the other methods which return property definitions